tests/dopamine/utils/threading_utils_test.py

```diff
@@ -0,0 +1,164 @@
+# coding=utf-8
+# Copyright 2018 The Dopamine Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for threading_utils.py."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import mock
+import tempfile
+import threading
+
+from dopamine.agents.dqn import dqn_agent
+from dopamine.utils import threading_utils
+import tensorflow as tf
+from tensorflow import test
+
+
+def _get_internal_name(attr):
+  thread_id = str(threading.current_thread().ident)
+  return '__' + attr + '_' + thread_id
+
+
+class ThreadsTest(test.TestCase):
+  """Unit tests for threading utils."""
+
+  def testDefaultValueIsAdded(self):
+    """Tests that the default value is properly set by the helper."""
+    obj = mock.Mock()
+    threading_utils.initialize_local_attributes(obj, attr=3)
+    self.assertEqual(obj.attr_default, 3)
+
+  def testMultipleDefaultValuesAreSet(self):
+    """Tests that multiple default values are properly set by the helper."""
+    obj = mock.Mock()
+    threading_utils.initialize_local_attributes(obj, attr1=3, attr2=4)
+    self.assertEqual(obj.attr1_default, 3)
+    self.assertEqual(obj.attr2_default, 4)
+
+  def testAttributeDefaultValueIsCalled(self):
+    """Tests that getter properly uses the default value."""
+    MockClass = threading_utils.local_attributes(['attr'])(
```

Comment:
Dang -- how did you figure this voodoo out?

Reply:
That's what the decorator does :) - `local_attributes` is called with the list of attributes in argument which creates a _decorator function, which is then called with a class in argument and returns the new class needed for the test.